py/dml/c_backend.py

```diff
@@ -917,120 +925,145 @@ def generate_implements(code, device):
         except DMLError as e:
             report(e)
 
-def simple_event_fun_deconstruct_data(method, data):
-    if method.dimensions > 0 and len(method.inp) > 0:
-        out(f'ASSERT(SIM_attr_list_size({data}) == 2);\n')
-        out(f'attr_value_t indices = SIM_attr_list_item({data}, 0);\n')
-        out(f'attr_value_t params = SIM_attr_list_item({data}, 1);\n')
-    elif method.dimensions > 0 or len(method.inp) > 0:
-        varname = 'indices' if method.dimensions > 0 else 'params'
-        out(f'attr_value_t {varname} = {data};\n')
-
-    if method.dimensions > 0:
-        out(f'ASSERT(SIM_attr_list_size(indices) == {method.dimensions});\n')
+def generate_simple_event_only_domains_funs():
+    start_function_definition(
+        'attr_value_t _simple_event_only_domains_get_value('
+        + 'conf_object_t *_obj, lang_void *data)')
+    out('{\n', postindent=1)
+    out('return _serialize_simple_event_data(0, NULL, _id_infos, '
+        + '(_simple_event_data_t *)data);\n')
+    out('}\n', preindent=-1)
+    splitting_point()
```

Comment:
`splitting_point()` is redundant here: it should be added after generating a function that can be large or plentiful in DML 1.2. The `--split-c-file` option is a workaround for the kind of performance problems you get in 1.2, and is largely irrelevant in 1.4.

Reply:
Noted, will remove.